### ISSUE TITLE
Testnet metashard indexing error [validatorsRating.rating] cannot be changed from type [float] to [long]

### ISSUE THREAD

**Issue by jun0tpyrc** [ElrondNetwork/elrond-go#2426]
```
could not index validators rating = error while parsing the response: code returned: 400, body: map[error:map[reason:mapper [validatorsRating.rating] cannot be changed from type [float] to [long]
```

<img width="1792" alt="Screenshot 2020-11-02 at 5 24 41 PM" src="https://user-images.githubusercontent.com/68580727/97851831-dc80db00-1d30-11eb-98fd-4cbf9afa7dcf.png">

**Comment by AdoAdoAdo** [comment]
@bogdan-rosianu can you please take a look?

**Comment by miiu96** [comment]
Hello. Can you tell me more about your configuration. You start a metachain observer with elasticsearch database right ? maybe you can tell me what version of elasticsearch server you use ?

**Comment by jun0tpyrc** [comment]
> Hello. Can you tell me more about your configuration. You start a metachain observer with elasticsearch database right ? maybe you can tell me what version of elasticsearch server you use ?

we use docker powered elasticsearch , image tag `docker.elastic.co/elasticsearch/elasticsearch:7.9.0`

**Comment by miiu96** [comment]
And also can tell me what release/branch do  you use in order to can reproduce your problem ? Thank you

**Comment by jun0tpyrc** [comment]
> And also can tell me what release/branch do you use in order to can reproduce your problem ? Thank you

we use the public image `elrondnetwork/elrond-go-node:v1.1.6`
which already asking for the elasticIndexTemplates when we turn on the ElasticSearchConnector plugin , so the config included the files from
https://github.com/ElrondNetwork/elrond-config-testnet/tree/master/elasticIndexTemplates 
too

**Comment by miiu96** [comment]
Thank you for the information I will take a look and come with feedback.

**Comment by LucianMincu** [comment]
Hi @jun0tpyrc! 

TL:DR I suggest to perform a complete cleanup on your ES db and restart the node instance starting right from v1.1.6.1 and there should no other problems there. 🙏 

Here's what happened:
1. The testnet config release [v1.1.6.0](https://github.com/ElrondNetwork/elrond-config-testnet/tree/T1.1.6.0) have not included the folder named `elasticIndexTemplates` and all the index configuration templates for the ES instance. 
2. The latest testnet config release [v1.1.6.1](https://github.com/ElrondNetwork/elrond-config-testnet/tree/T1.1.6.1) have been released exactly for this particular reason shortly after the previous version where we have added the `elasticIndexTemplates` and all the index configuration templates as well. 
3. Because you started indexing data into the ES with the `v1.1.6.0` (without the configuration templates) and the than rolled the upgrade to `v1.1.6.1` the node instance was not able to write any data into the previous indexes from ES as various settings have been applied such as index-naming changes from `transactions` to `transactions-00001` in preparation for a long term solution for rotating indexes if the average throughput increases and so the size of each index inside the ES.

Thanks and looking forward if this was the case for you. 🙏 👍

**Comment by jun0tpyrc** [comment]
I can confirm it looks good when ES data cleaned and rerun to create new indices starting current epoch 

It looks the version for 1.1.6.1 release is not very semantic(for those using elasticsearch connector, because old data can't be retained and queried this way), but understandable for keeping ongoing performance scaling up  

Thanks for the explanation 🙏

**Comment by jun0tpyrc** [comment]
but we are querying elasticsearch ourselves for product ,  may i confirm the logic when the nodes will start to write to indices like `blocks-000001` instead of `blocks`?   @LucianMincu 
we might need to reconsider our implementation if we need to consider this too

**Comment by jun0tpyrc** [comment]
> Thanks for the explanation 🙏
> It looks the version for 1.1.6.1 release is not very semantic(for those using elasticsearch connector, because old data can't be retained and queried this way), but understandable for keeping ongoing performance scaling up
> 
> seeing this again after running a few hours , might be erasing ES alone is not enough , I need to examine again
> nodes running on
> `elrondnetwork/elrond-go-node:devlatest@sha256:b3ebaf3cc29c0958a874ab06f22285d518fdfcf533e7e6d142b7708423548fc6`

it occurs again when going into new epoch for metashard 

```
^[[0;32mINFO ^[[0m[2020-11-05 05:13:57.492]   meta block has been committed successfully ^[[0;32mepoch^[[0m = 142 ^[[0;32mround^[[0m = 343339 ^[[0;32mnonce^[[0m = 342337 ^[[0;32mhash^[[0m = 668411bdec4341ae66324eb909781104a55cc59174e1363b179fc9d0e6e1d47f
^[[0;32mINFO ^[[0m[2020-11-05 05:14:03.563]   meta block has been committed successfully ^[[0;32mepoch^[[0m = 142 ^[[0;32mround^[[0m = 343340 ^[[0;32mnonce^[[0m = 342338 ^[[0;32mhash^[[0m = 32fcbed49ec600bf2b43c237a983b27e5ab5c7adbdbd6c9de86726989e468a91
^[[0;32mINFO ^[[0m[2020-11-05 05:14:15.494]   meta block has been committed successfully ^[[0;32mepoch^[[0m = 142 ^[[0;32mround^[[0m = 343342 ^[[0;32mnonce^[[0m = 342339 ^[[0;32mhash^[[0m = ce0c2746d1eb861cd0a938e03c5856562b156061a94aee504fa572082f79a5fa
^[[0;33mWARN ^[[0m[2020-11-05 05:14:21.493]   Did not find nodesConfig                 ^[[0;33mepoch^[[0m = 143
^[[0;32mINFO ^[[0m[2020-11-05 05:14:21.532]   meta block has been committed successfully ^[[0;32mepoch^[[0m = 143 ^[[0;32mround^[[0m = 343343 ^[[0;32mnonce^[[0m = 342340 ^[[0;32mhash^[[0m = 0a3fe022d1ec1f22f093ee30ff78934d3fef9f20c080e565835ec3ed982ef4ae
^[[0;33mWARN ^[[0m[2020-11-05 05:14:22.079]   itemRating.Save                          ^[[0;33mcould not index validators rating^[[0m = error while parsing the response: code returned: 400, body: map[error:map[reason:mapper [validatorsRating.rating] cannot be changed from type [long] to [float] root_cause:[map[reason:mapper [validatorsRating.rating] cannot be changed from type [long] to [float] type:illegal_argument_exception]] type:illegal_argument_exception] status:400], bodyBytes: [123 34 101 114 114 111 114 34 58 123 34 114 111 111 116 95 99 97 117 115 101 34 58 91 123 34 116 121 112 101 34 58 34 105 108 108 101 103 97 108 95 97 114 103 117 109 101 110 116 95 101 120 99 101 112 116 105 111 110 34 44 34 114 101 97 115 111 110 34 58 34 109 97 112 112 101 114 32 91 118 97 108 105 100 97 116 111 114 115 82 97 116 105 110 103 46 114 97 116 105 110 103 93 32 99 97 110 110 111 116 32 98 101 32 99 104 97 110 103 101 100 32 102 114 111 109 32 116 121 112 101 32 91 108 111 110 103 93 32 116 111 32 91 102 108 111 97 116 93 34 125 93 44 34 116 121 112 101 34 58 34 105 108 108 101 103 97 108 95 97 114 103 117 109 101 110 116 95 101 120 99 101 112 116 105 111 110 34 44 34 114 101 97 115 111 110 34 58 34 109 97 112 112 101 114 32 91 118 97 108 105 100 97 116 111 114 115 82 97 116 105 110 103 46 114 97 116 105 110 103 93 32 99 97 110 110 111 116 32 98 101 32 99 104 97 110 103 101 100 32 102 114 111 109 32 116 121 112 101 32 91 108 111 110 103 93 32 116 111 32 91 102 108 111 97 116 93 34 125 44 34 115 116 97 116 117 115 34 58 52 48 48 125]
^[[0;33mWARN ^[[0m[2020-11-05 05:14:22.079]   dataDispatcher.doWork could not index item (will retry) ^[[0;33merror^[[0m = error while parsing the response: code returned: 400, body: map[error:map[reason:mapper [validatorsRating.rating] cannot be changed from type [long] to [float] root_cause:[map[reason:mapper [validatorsRating.rating] cannot be changed from type [long] to [float] type:illegal_argument_exception]] type:illegal_argument_exception] status:400], bodyBytes: [123 34 101 114 114 111 114 34 58 123 34 114 111 111 116 95 99 97 117 115 101 34 58 91 123 34 116 121 112 101 34 58 34 105 108 108 101 103 97 108 95 97 114 103 117 109 101 110 116 95 101 120 99 101 112 116 105 111 110 34 44 34 114 101 97 115 111 110 34 58 34 109 97 112 112 101 114 32 91 118 97 108 105 100 97 116 111 114 115 82 97 116 105 110 103 46 114 97 116 105 110 103 93 32 99 97 110 110 111 116 32 98 101 32 99 104 97 110 103 101 100 32 102 114 111 109 32 116 121 112 101 32 91 108 111 110 103 93 32 116 111 32 91 102 108 111 97 116 93 34 125 93 44 34 116 121 112 101 34 58 34 105 108 108 101 103 97 108 95 97 114 103 117 109 101 110 116 95 101 120 99 101 112 116 105 111 110 34 44 34 114 101 97 115 111 110 34 58 34 109 97 112 112 101 114 32 91 118 97 108 105 100 97 116 111 114 115 82 97 116 105 110 103 46 114 97 116 105 110 103 93 32 99 97 110 110 111 116 32 98 101 32 99 104 97 110 103 101 100 32 102 114 111 109 32 116 121 112 101 32 91 108 111 110 103 93 32 116 111 32 91 102 108 111 97 116 93 34 125 44 34 115 116 97 116 117 115 34 58 52 48 48 125]
^[[0;32mINFO ^[[0m[2020-11-05 05:14:24.404]   meta block has been committed successfully ^[[0;32mepoch^[[0m = 143 ^[[0;32mround^[[0m = 343344 ^[[0;32mnonce^[[0m = 342341 ^[[0;32mhash^[[0m = 90efe07a1ce8edcca597d6ad87492e29a7f965e4af7f52e5fd804f8b2be40be5
^[[0;33mWARN ^[[0m[2020-11-05 05:14:25.091]   itemRating.Save                          ^[[0;33mcould not index validators rating^[[0m = error while parsing the response: code returned: 400, body: map[error:map[reason:mapper [validatorsRating.rating] cannot be changed from type [long] to [float] root_cause:[map[reason:mapper [validatorsRating.rating] cannot be changed from type [long] to [float] type:illegal_argument_exception]] type:illegal_argument_exception] status:400], bodyBytes: [123 34 101 114 114 111 114 34 58 123 34 114 111 111 116 95 99 97 117 115 101 34 58 91 123 34 116 121 112 101 34 58 34 105 108 108 101 103 97 108 95 97 114 103 117 109 101 110 116 95 101 120 99 101 112 116 105 111 110 34 44 34 114 101 97 115 111 110 34 58 34 109 97 112 112 101 114 32 91 118 97 108 105 100 97 116 111 114 115 82 97 116 105 110 103 46 114 97 116 105 110 103 93 32 99 97 110 110 111 116 32 98 101 32 99 104 97 110 103 101 100 32 102 114 111 109 32 116 121 112 101 32 91 108 111 110 103 93 32 116 111 32 91 102 108 111 97 116 93 34 125 93 44 34 116 121 112 101 34 58 34 105 108 108 101 103 97 108 95 97 114 103 117 109 101 110 116 95 101 120 99 101 112 116 105 111 110 34 44 34 114 101 97 115 111 110 34 58 34 109 97 112 112 101 114 32 91 118 97 108 105 100 97 116 111 114 115 82 97 116 105 110 103 46 114 97 116 105 110 103 93 32 99 97 110 110 111 116 32 98 101 32 99 104 97 110 103 101 100 32 102 114 111 109 32 116 121 112 101 32 91 108 111 110 103 93 32 116 111 32 91 102 108 111 97 116 93 34 125 44 34 115 116 97 116 117 115 34 58 52 48 48 125]
^[[0;33mWARN ^[[0m[2020-11-05 05:14:25.091]   dataDispatcher.doWork could not index item (will retry) ^[[0;33merror^[[0m = error while parsing the response: code returned: 400, body: map[error:map[reason:mapper [validatorsRating.rating] cannot be changed from type [long] to [float] root_cause:[map[reason:mapper [validatorsRating.rating] cannot be changed from type [long] to [float] type:illegal_argument_exception]] type:illegal_argument_exception] status:400], bodyBytes: [123 34 101 114 114 111 114 34 58 123 34 114 111 111 116 95 99 97 117 115 101 34 58 91 123 34 116 121 112 101 34 58 34 105 108 108 101 103 97 108 95 97 114 103 117 109 101 110 116 95 101 120 99 101 112 116 105 111 110 34 44 34 114 101 97 115 111 110 34 58 34 109 97 112 112 101 114 32 91 118 97 108 105 100 97 116 111 114 115 82 97 116 105 110 103 46 114 97 116 105 110 103 93 32 99 97 110 110 111 116 32 98 101 32 99 104 97 110 103 101 100 32 102 114 111 109 32 116 121 112 101 32 91 108 111 110 103 93 32 116 111 32 91 102 108 111 97 116 93 34 125 93 44 34 116 121 112 101 34 58 34 105 108 108 101 103 97 108 95 97 114 103 117 109 101 110 116 95 101 120 99 101 112 116 105 111 110 34 44 34 114 101 97 115 111 110 34 58 34 109 97 112 112 101 114 32 91 118 97 108 105 100 97 116 111 114 115 82 97 116 105 110 103 46 114 97 116 105 110 103 93 32 99 97 110 110 111 116 32 98 101 32 99 104 97 110 103 101 100 32 102 114 111 109 32 116 121 112 101 32 91 108 111 110 103 93 32 116 111 32 91 102 108 111 97 116 93 34 125 44 34 115 116 97 116 117 115 34 58 52 48 48 125]
```

our storage pruning setup now
```
[StoragePruning]
   # If the Enabled flag is set to false, then the storers won't divide epochs into separate dbs
   Enabled = true

   # If this flag is set to false, the node won't delete any database between epochs
   CleanOldEpochsData = false

   # NumEpochsToKeep - if the flag above is set to true, this will set the number of epochs to keep in the storage.
   # Epochs older that (current epoch - NumOfEpochsToKeep) will be removed
   NumEpochsToKeep = 999

   # NumActivePersisters - this will set the number of persisters to keep active at a moment. This works even if
   # the node removes old epochs data or not. In case of a node which removes old epochs data, this value has to be
   # smaller or equal to the NumOfEpochsToKeep flag
   NumActivePersisters = 3
```

**Comment by jun0tpyrc** [comment]
or do we mean we'll have a image for v1.1.6.1 , but not starting `v1.1.6 image + 1.1.6.1 config + empty ES`?

**Comment by LucianMincu** [comment]
Hi @jun0tpyrc

> or do we mean we could have an image for v1.1.6.1 , but not starting `v1.1.6 elrond-node-go image + 1.1.6.1 config + empty-ES`?
Just to confirm, you already tried ` v1.1.6 elrond-node-go image + 1.1.6.1 config + empty-ES ` and after epoch change the same problem occurs, right?

 > `UseKibana         = false` 
The default ES installation does not provide by default he same kibana path so it should remain `false`. 

Can you point us to the docker image or build used for ES setup?
Also maybe a short archive with the entire configuration file of your node, removing any credentials or sensible data from your configs, sent over to us would help us reproduce the error. 

Thanks! 🙏

**Comment by miiu96** [comment]
I think we found your problem I need just a confirmation from you. Can you please provide me what this URL return on you elasticsearch db  URL: http:{YOUR_ELASTIC_SEARCH_SERVER}/rating ?

**Comment by jun0tpyrc** [comment]
> Hi @jun0tpyrc
> 
> > or do we mean we could have an image for v1.1.6.1 , but not starting `v1.1.6 elrond-node-go image + 1.1.6.1 config + empty-ES`?
> 
> Just to confirm, you already tried `v1.1.6 elrond-node-go image + 1.1.6.1 config + empty-ES` and after epoch change the same problem occurs, right?
> 
> > `UseKibana         = false`
> 
> The default ES installation does not provide by default he same kibana path so it should remain `false`.
> 
> Can you point us to the docker image or build used for ES setup?
> Also maybe a short archive with the entire configuration file of your node, removing any credentials or sensible data from your configs, sent over to us would help us reproduce the error.
> 
> Thanks! 🙏

For the latest env i seeing this , we are using 
```
elrondnetwork/elrond-go-node:devlatest DIGEST:sha256:b3ebaf3cc29c0958a874ab06f22285d518fdfcf533e7e6d142b7708423548fc6 
```
OS/ARCH linux/amd64 SIZE 57.61 MB LAST PUSHED 16 hours ago by elrondnetwork

I thought judging from the push datetime it is the newest, but it looks strange here for being 1.15 something
<img width="621" alt="Screenshot 2020-11-05 at 4 58 24 PM" src="https://user-images.githubusercontent.com/68580727/98219571-257e9c80-1f88-11eb-9b90-a8d55d6aa743.png">

can we confirm what version should this devlatest version be pointing at ?

**Comment by jun0tpyrc** [comment]
> I think we found your problem I need just a confirmation from you. Can you please provide me what this URL return on you elasticsearch db URL: http:{YOUR_ELASTIC_SEARCH_SERVER}/rating ? @jun0tpyrc

```
{
  "rating": {
    "aliases": {},
    "mappings": {},
    "settings": {
      "index": {
        "creation_date": "1604553261971",
        "number_of_shards": "1",
        "number_of_replicas": "1",
        "uuid": "Jio7DgIlT46CwQYb7xtaGA",
        "version": {
          "created": "7090099"
        },
        "provided_name": "rating"
      }
    }
  }
}
```

**Comment by miiu96** [comment]
Hello @jun0tpyrc a PR that should solve your problem was merged in our development branch. Please try again with  `development` branch if is everything ok now.  Thank you.

**Comment by miiu96** [comment]
Hello @jun0tpyrc
Can we close this issue ? Was solved your problem  ? Thank you. :+1:

**Comment by jun0tpyrc** [comment]
> Hello @jun0tpyrc
> Can we close this issue ? Was solved your problem ? Thank you. 👍

yes, it looks working well for testnet over last few epoches , we might close this first ,  thanks ~
